src/qmapshack/gis/wpt/CGisItemWpt.h

```diff
@@ -348,7 +348,7 @@ class CGisItemWpt : public IGisItem
     const searchValue_t getValueByKeyword(searchProperty_e keyword) override;
 
     static QString getLastName(const QString &name);
-    static void newWpt(const QPointF &pt, const QString& name, const QString& desc, IGisProject *project, const QString &icon = "", bool openEditWIndow = true);
+    static void newWpt(const QPointF &pt, const QString& name, const QString& desc, IGisProject *project, const QString &icon = "", const QList<link_t> links = QList<link_t>(), quint32 ele = NOINT, bool openEditWIndow = true);
```

Comment:
Instead of passing this endless list of parameters and default arguments wouldn't it be better to override newWpt and pass the POI object?